src/SoftRobots/component/camera/PointCloudProcessing.cpp

```diff
@@ -0,0 +1,537 @@
+    //build rotation matrix
+    Eigen::Matrix4f transform_1 = Eigen::Matrix4f::Identity();
+    // Define a rotation matrix (see https://en.wikipedia.org/wiki/Rotation_matrix)
+    transform_1 (0,0) = -0.69419594;
+    transform_1 (0,1) = -0.71476897;
+    transform_1 (0,2) = 0.08483703;
+    transform_1 (1,0) = -0.7196964;
+    transform_1 (1,1) = 0.68741082;
+    transform_1 (1,2) = -0.09748567;
+    transform_1 (2,0) = 0.01136184;
+    transform_1 (2,1) = -0.12873106;
+    transform_1 (2,2) = -0.99161445;
+    // Define a translation.
+    transform_1 (0,3) = x0/1000;
+    transform_1 (1,3) = y0/1000;
+    transform_1 (2,3) = z0/1000;
+
+
+    //////////////////////////////////////////////////////////////////Image segmentation/////////////////////////////////////////////////
+    //Color-based region growing segmentation
+    pcl::search::KdTree<pcl::PointXYZRGB>::Ptr tree (new pcl::search::KdTree<pcl::PointXYZRGB>);
+    //This line is responsible for pcl::RegionGrowingRGB instantiation. This class has two parameters:
+    //PointT - type of points to use(in the given example it is pcl::PointXYZRGB)
+    //NormalT - type of normals to use. Insofar as pcl::RegionGrowingRGB is derived from the pcl::RegionGrowing, it can use both tests at the same time: color test and normal test. The given example uses only the first one, therefore type of normals is not used.
+    pcl::RegionGrowingRGB<pcl::PointXYZRGB> reg;
+    //These lines provide the instance with the input cloud, and search method.
+    reg.setInputCloud (filteredpoints);
+    reg.setSearchMethod (tree);
+    // It is used to determine whether the point is neighbouring or not.
+    // If the point is located at a distance less than the given threshold, then it is considered to be neighbouring.
+    // It is used for clusters neighbours search.
+    reg.setDistanceThreshold (0.01);
+    //This line sets the color threshold.
+    //Just as angle threshold is used for testing points normals in pcl::RegionGrowing to determine if the point belongs to cluster, this value is used for testing points colors.
+    reg.setPointColorThreshold (6);
+    //Here the color threshold for clusters is set. This value is similar to the previous, but is used when the merging process takes place.
+    reg.setRegionColorThreshold (4);
+    //it is used for merging process mentioned in the beginning. If cluster has less points than was set through setMinClusterSize method, then it will be merged with the nearest neighbour.
+    reg.setMinClusterSize (100); //set the size of the class
+
+    //Here is the place where the algorithm is launched. It will return the array of clusters when the segmentation process will be over.
+    std::vector <pcl::PointIndices> clusters;
+    reg.extract (clusters);
+
+    std::cout << "Number of clusters is equal to " << clusters.size () << std::endl;
+    std::cout << "First cluster has " << clusters[0].indices.size () << " points." << endl;
+
```

Comment:
I feel like this contribution could be cleaned during a Friday coding session...? maybe